From 1d2694763ecec095a1358b46e50b54fea8a98469 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Tue, 20 Sep 2016 19:54:41 +0800
Subject: [PATCH 2/5] fs: yaffs2: switch to the new prototype for
 ->set/getxattr()

Since commit 3767e255b390 ("switch ->setxattr() to passing dentry and
inode separately") and commit ce23e6401334 ("->getxattr(): pass dentry
and inode as separate arguments"), a new inode argument has been added
for these two functions. Adjust these two functions to accommodate to
this change.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 fs/yaffs2/yaffs_vfs.c | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/fs/yaffs2/yaffs_vfs.c b/fs/yaffs2/yaffs_vfs.c
index 9aa007d9ded6..cd387d7abfa2 100644
--- a/fs/yaffs2/yaffs_vfs.c
+++ b/fs/yaffs2/yaffs_vfs.c
@@ -919,10 +919,10 @@ static int yaffs_setattr(struct dentry *dentry, struct iattr *attr)
 	return error;
 }
 
-static int yaffs_setxattr(struct dentry *dentry, const char *name,
-		   const void *value, size_t size, int flags)
+static int yaffs_setxattr(struct dentry *dentry, struct inode *inode,
+			  const char *name, const void *value, size_t size,
+			  int flags)
 {
-	struct inode *inode = dentry->d_inode;
 	int error = 0;
 	struct yaffs_dev *dev;
 	struct yaffs_obj *obj = yaffs_inode_to_obj(inode);
@@ -952,10 +952,9 @@ static int yaffs_setxattr(struct dentry *dentry, const char *name,
 	return error;
 }
 
-static ssize_t yaffs_getxattr(struct dentry * dentry, const char *name,
-			void *buff, size_t size)
+static ssize_t yaffs_getxattr(struct dentry * dentry, struct inode *inode,
+			      const char *name, void *buff, size_t size)
 {
-	struct inode *inode = dentry->d_inode;
 	int error = 0;
 	struct yaffs_dev *dev;
 	struct yaffs_obj *obj = yaffs_inode_to_obj(inode);
-- 
2.5.0

